client/ayon_core/hosts/max/plugins/publish/validate_mesh_has_uv.py

```diff
@@ -0,0 +1,44 @@
+
+import pyblish.api
+from ayon_core.hosts.max.api.action import SelectInvalidAction
+from ayon_core.pipeline.publish import (
+    ValidateMeshOrder,
+    OptionalPyblishPluginMixin,
+    PublishValidationError
+)
+from pymxs import runtime as rt
+
+
+class ValidateMeshHasUVs(pyblish.api.InstancePlugin,
+                         OptionalPyblishPluginMixin):
+
+    """Validate the current mesh has UVs.
+
+    It validates whether the current mesh has texture vertex(s).
+    If the mesh does not have texture vertex(s), it does not
+    have UVs in Max.
+    """
+
+    order = ValidateMeshOrder
+    hosts = ['max']
+    families = ['model']
+    label = 'Validate Mesh Has UVs'
+    actions = [SelectInvalidAction]
+    optional = True
+
+
+    @classmethod
+    def get_invalid(cls, instance):
+        invalid = [member for member in instance.data["members"]
+                   if not member.mesh.numTVerts > 0]
```

Comment:
Having no knowledge of Max - but in Maya it's possible to have a mesh that has only partial UVs, e.g. a few faces missing UVs while the rest does have UVs. Is that possible in Max as well?

If so, we might want to clarify in the wording that it currently only checks if an object has ANY uvs and not whether all faces have UVs.

Reply:
it doesn't really have the faces missing UVs check. there is a function to get the coordinates of texture faces but there is no exact function to get number of texture faces(but it does have the function to get number of face)
https://help.autodesk.com/view/MAXDEV/2022/ENU/?guid=GUID-2E5C8194-F165-4D09-8A19-069BFCF83104